sql/src/planner.rs

```diff
@@ -380,6 +383,15 @@ impl<'a, P: MetaProvider> PlannerDelegate<'a, P> {
 
         // Timestamp column must be provided.
         let timestamp_name = timestamp_name.context(RequireTimestamp)?;
+        // The timestamp key column must not be a Tag column
+        if name_column_map.get(&timestamp_name as &str).is_some() {
+            ensure!(
+                !name_column_map.get(&timestamp_name as &str).unwrap().is_tag,
+                TimestampColumnCanNotTag {
+                    name: &timestamp_name as &str,
+                }
+            )
+        }
```

Comment:
Maybe we can put it like:
```suggestion
        if let Some(timestamp_column) = name_column_map.get(&timestamp_name) {
            ensure!(
                !timestamp_column.is_tag,
                TimestampColumnCanNotTag {
                    name: &timestamp_name,
                }
            )
        }
```

Reply:
OK I will. Thanks